src/renderer/components/settings/WindowZoom.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0
+ * Vesktop, a desktop app aiming to give you a snappier Discord Experience
+ * Copyright (c) 2023 Vendicated and Vencord contributors
+ */
+
+import { Margins } from "@vencord/types/utils";
+import { Forms, Slider, useEffect, useState } from "@vencord/types/webpack/common";
+
+import { SettingsComponent } from "./Settings";
+
+export const WindowZoom: SettingsComponent = ({ settings }) => {
+    const [zoomFactor, setZoomFactor] = useState(settings.zoomFactor ?? 1);
+
+    useEffect(() => {
+        const handleZoomChange = event => {
+            console.log("zoom changed", event.detail);
+            setZoomFactor(event.detail);
+        };
+
+        window.addEventListener("zoomChanged", handleZoomChange);
+
+        return () => {
+            window.removeEventListener("zoomChanged", handleZoomChange);
+        };
+    }, []);
+
+    return (
+        <>
+            <Forms.FormTitle className={Margins.top16 + " " + Margins.bottom8}>Zoom Level</Forms.FormTitle>
+            <Slider
+                className={Margins.top20}
+                initialValue={zoomFactor}
+                defaultValue={1}
+                onValueChange={v => {
+                    settings.zoomFactor = v;
+                    VesktopNative.win.zoom(v);
+                    setZoomFactor(v);
```

Comment:
you don't need this state. via `useSettings()`, you can get a stateful settings object that will automatically rerender your component on change

the parent component already uses this hook so you should not need to call it yourself